docs/mesos/Dockerized27Deployment.md

```diff
@@ -0,0 +1,226 @@
+
+## Docker
+Since this is a dockerized deployment, you will need Docker installed on every Master and Agent in your cluster.
+[Follow Docker's Centos installation guide](https://docs.docker.com/engine/installation/centos/) for information on how to get Docker installed.
+
+## FQDN
+These instructions assume each host can reach other hosts using their fully qualified domain names (FQDN).  To check the FQDN on a host use
+
+    hostname -f
+
+Then attempt to ping that name from other servers.
+
+Also important are that Calico and Mesos have the same view of the (non-fully-qualified) hostname.  In particular, the value returned by
+
+    hostname
+
+must be unique for each node in your cluster.  Both Calico and Mesos use this value to identify the host.
+
+## SSL
+The Marathon build we'll be using requires SSL enabled in Mesos on each Master and Slave in order to pull docker images from dockerhub. The systemd services we'll be using in this guide are already configured to search for the appropriate SSL keys in `/certs`. Run the following commands on each Master and slave to place those certs:
+
+      sudo mkdir /keys
+      sudo openssl genrsa -f4  -out /keys/key.pem 4096
+      sudo openssl req -new -batch -x509  -days 365 -key /keys/key.pem -out /keys/cert.pem
+
+# Prepare Master
+Perform the following steps on one host which will be designated the "Master".
+
+## 1. Configure your firewall
+
+You will either need to configure the firewalls on each node in your cluster (recommended) to allow access to the cluster services or disable it completely.  Included in this section is configuration examples for `firewalld`.
+
+Master node(s) require
+
+| Service Name | Port/protocol     |
+|--------------|-------------------|
+| zookeeper    | 2181/tcp          |
+| mesos-master | 5050/tcp          |
+| etcd         | 2379/tcp 4001/tcp |
+| marathon     | 8080/tcp          |
+
+Example `firewalld` config
+
+    sudo firewall-cmd --zone=public --add-port=2181/tcp --permanent
+    sudo firewall-cmd --zone=public --add-port=5050/tcp --permanent
+    sudo firewall-cmd --zone=public --add-port=2379/tcp --permanent
+    sudo firewall-cmd --zone=public --add-port=4001/tcp --permanent
+    sudo firewall-cmd --zone=public --add-port=8080/tcp --permanent
+    sudo systemctl restart firewalld
+    
+# 2. Zookeeper
+Download the Zookeeper image, as well as the systemd service which will ensure Zookeeper is kept running:
+
+    sudo docker pull jplock/zookeeper:3.4.5
+    wget https://raw.githubusercontent.com/projectcalico/calico-mesos/0.27/dockerized-mesos/config/units/zookeeper.service
+    sudo cp zookeeper.service /usr/lib/systemd/system/
+    sudo systemctl enable zookeeper.service
+    sudo systemctl start zookeeper.service
```

Comment:
We should prompt to check status after or do some kind of sanity check that each step is working as expected.